src/main/java/com/noxag/newnox/textanalyzer/algorithms/BibliographyAnalyzer.java

```diff
@@ -1,18 +1,111 @@
+        try {
+            findings.addAll(getReferencesWithoutSource(doc));
+        } catch (IOException e) {
+            LOGGER.log(Level.WARNING, "Could not strip text from document", e);
+        }
+        return findings;
+    }
+
+    private List<Finding> getReferencesWithoutSource(PDDocument doc) throws IOException {
+        List<Finding> findings = new ArrayList<>();
+        List<PDFPage> pages = PDFTextExtractionUtil.extractText(doc);
+        List<String> bibliographyHints = readBibliographyIdentificationListFile(BIBLIOGRAPHY_IDENTIFICATION_LIST_PATH);
+        pages.stream().forEach(page -> {
+            TextPositionSequence firstWordOfPage = page.getFirstWord();
+            if (bibliographyHints.stream().anyMatch(firstWordOfPage.toString().toLowerCase()::contains)) {
+                List<TextPositionSequence> markedSources = getAllSources(pages, firstWordOfPage.getPageIndex());
+                List<TextPositionSequence> referencedSources = getAllReferences(pages, firstWordOfPage.getPageIndex());
+                findings.addAll(compareReferencesWithSources(referencedSources, markedSources));
+            }
+        });
+        return findings;
+    }
+
+    // Gets all references to sources of content pages
+    private List<TextPositionSequence> getAllReferences(List<PDFPage> pages, int sourcePageIndex) {
+        List<PDFPage> contentPages = pages.stream().filter(page -> page.getPageIndex() < sourcePageIndex)
+                .collect(Collectors.toList());
+        return getSourceList(contentPages);
+    }
+
+    // Gets all sources in bibliography
+    private List<TextPositionSequence> getAllSources(List<PDFPage> pages, int sourcePageIndex) {
+        List<PDFPage> sourcePages = pages.stream().filter(page -> page.getPageIndex() >= sourcePageIndex)
+                .collect(Collectors.toList());
+        return getSourceList(sourcePages);
+    }
+
+    // Reads all sources/references out of given list of PDFPages
+    private List<TextPositionSequence> getSourceList(List<PDFPage> pageList) {
+        List<TextPositionSequence> wordsOfBibliography = PDFTextExtractionUtil.extractWords(pageList);
+        List<TextPositionSequence> markedSources = new ArrayList<>();
+        wordsOfBibliography.stream().forEach(word -> {
+            if (word.toString().contains("[") && word.toString().contains("]")) {
+                markedSources.add(word);
+            }
+        });
+        return markedSources;
+    }
+
+    // Compares, if reference has been stored in bibliography
+    private List<Finding> compareReferencesWithSources(List<TextPositionSequence> referencedSources,
+            List<TextPositionSequence> markedSources) {
+        List<Finding> findings = new ArrayList<>();
+        referencedSources.stream().forEach(referencedSource -> {
+            // Reads reference part out of whole word
+            String reference = referencedSource.toString().substring(referencedSource.toString().indexOf("["),

```

Comment:
zur besseren lesbarkeit könntest du diese methoden-kette in eine eine Methode auslagern und ihr einen aussagekräftigen namen geben, ist aber optional

Reply:
done